### ISSUE TITLE
function to spread false positive

### ISSUE THREAD

**Issue by reedy** [hammer-tools/php-hammer#24]
Not the original reporter of https://plugins.jetbrains.com/plugin/19515-php-hammer/reviews#review=87175 but primarily working on a PHP 7.4 (min) project, this is relevant too

```php
<?php

$foo = [ 'foo'=> 'bar'];
$bar = [ 'bar' => 'foo' ];

$merged = array_merge( $foo, $bar );
```

![Screenshot 2023-08-22 at 20 05 07](https://github.com/hammer-tools/php-hammer/assets/67615/ba50f0bd-134d-490f-aa15-4e12f5995873)

if we replace it with

```php
<?php

$foo = [ 'foo'=> 'bar'];
$bar = [ 'bar' => 'foo' ];

$merged = [...$foo, ...$bar];
```

PHPStorm does complain

![Screenshot 2023-08-22 at 20 06 31](https://github.com/hammer-tools/php-hammer/assets/67615/17a70b98-53a2-46a9-b15e-827ce4d5e2f1)

as per https://php.watch/versions/8.1/spread-operator-string-array-keys

>    Since PHP 7.4, PHP supports array spread operator ( ...) for array unpacking. An array can be unpacked with another array if the array expression is prefixed with the spread operator. This effectively results in an array_merge of all arrays in the expression.

and

>    Prior to PHP 8.1, the spread operator only supported arrays with numeric keys. Using an array with string key resulted in an error prior to PHP 8.1:

A "simple" fix would be to potentially skip this check if the "PHP Language Level" in the IDE is set to < PHP 8.1...

A more complex fix would be looking for string keys (which gets harder in more dynamic code etc).

**Comment by rentalhost** [comment]
Thanks!

Maybe I can do a parsing on the variables and look for strings: if found, apply to 8.1+ only. If unable to detect (dynamic codes) then accept only for 8.1+ as well.

**Comment by rentalhost** [comment]
Thank you for taking the time to report this issue. The fix has been implemented in version **0.24.0** and is available on the **[Releases](https://github.com/hammer-tools/php-hammer/releases)** page for manual installation. Alternatively, you can wait for the official release on the **JetBrains Marketplace**, which should happen in approximately 3 days.

- **Concerning the issue you pointed out:** I've restricted the suggestion of using the spread operator to PHP 8.1 and later. However, some simpler cases may still work in earlier versions, especially in situations where it's guaranteed that there won't be any issues.
